### DISCUSSION TITLE
Setting WIDTH and HEIGHT for images causes image shrinking with padding/borders

### DISCUSSION THREAD

**Discussion by Jamie-99a** [twbs/bootstrap#40863] [Q&A]
In order to reduce content shifting on page loading with several embedded images, many are now advising we go retro and add explicit WIDTH and HEIGHT attributes to IMG tags for responsive images.

The problem when doing this with Bootstrap is that as soon as we introduce any padding or borders to the images, the WIDTH and HEIGHT attributes end up defining maximum size of the image **plus** the padding/borders, rather than just the image itself, hence shrinking the image to satisfy the WIDTH and HEIGHT attributes.

This does not occur when defining margins - only when introducing padding and borders.

Refer to the screenshot included below

![screenshot](https://github.com/user-attachments/assets/8efb8374-09f2-4579-804a-267c9b2b4393)

The simple code to test this is below.

Is this a Bootstrap bug, or should I be doing things differently?

```html
<!DOCTYPE HTML>
<HTML>

<HEAD>
<link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/css/bootstrap.min.css" rel="stylesheet">
</HEAD>

<BODY>

<IMG SRC="https://i.ibb.co/bm4WN1g/bp.png" CLASS="img-fluid border border-primary p-2 m-2" ALT="BP">
<BR>

<IMG SRC="https://i.ibb.co/bm4WN1g/bp.png" WIDTH="45" HEIGHT="60" CLASS="img-fluid border border-primary p-2 m-2" ALT="BP">
<BR>

</BODY>
</HTML>
```

**Comment by patkon** [comment]
I think the classes for padding and margin `(p-* and m-*)` are not intended for images. In your case, it is probably the interaction with `img-fluid` that causes the effect.

The img-fluid class causes the image to be set to max-width: 100%. And so the browser ignores the width/height specifications of the image.

**Comment by ynng3** [comment]
Adding padding or border to an image is not a good choice because the border and padding size can affect the image size.

Please change it as follows.
```html
<!DOCTYPE HTML>
<HTML>

<HEAD>
<link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/css/bootstrap.min.css" rel="stylesheet">
</HEAD>

<BODY>

<DIV CLASS="d-inline-block border border-primary p-2 m-2">
<IMG SRC="https://i.ibb.co/bm4WN1g/bp.png" CLASS="img-fluid" ALT="BP">
</DIV>

<BR>

<DIV CLASS="d-inline-block border border-primary p-2 m-2">
<IMG SRC="https://i.ibb.co/bm4WN1g/bp.png" WIDTH="45" HEIGHT="60" CLASS="img-fluid" ALT="BP">
</DIV>
<BR>

</BODY>
</HTML>
```

**Reply by Jamie-99a** [reply]
Thanks for your contribution. The use of an inline-block DIV wrapping around the image correctly works to preserve the explicit image height/width attributes, with the image border and padding styling moved to the surrounding DIV.
